specification/datacollaboration/resource-manager/Microsoft.DataCollaboration/preview/2020-05-04-preview/DataCollaboration.json

```diff
@@ -0,0 +1,5736 @@
+            "description": "continuation token",
+            "required": false,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DataAssetReferenceList"
+            }
+          },
+          "default": {
+            "description": "An error response received from the Microsoft.DataCollaboration resource provider.",
+            "schema": {
+              "$ref": "#/definitions/DataShareError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "DataAssetReferences_ListByProposal": {
+            "$ref": "./examples/DataAssetReferences_ListByProposal.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataCollaboration/workspaces/{workspaceName}/proposals/{proposalName}/dataAssetReferences/{referenceName}": {
+      "get": {
+        "tags": [
+          "DataAssetReference"
+        ],
+        "summary": "Get a DataAssetReference in a Proposal.",
+        "description": "Get a DataAssetReference in a Proposal",
+        "operationId": "DataAssetReferences_Get",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/workspaceName"
+          },
+          {
+            "name": "proposalName",
+            "in": "path",
+            "description": "The name of the proposal.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "referenceName",
+            "in": "path",
+            "description": "The name of the reference.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DataAssetReference"
+            }
+          },
+          "default": {
+            "description": "An error response received from the Microsoft.DataCollaboration resource provider.",
+            "schema": {
+              "$ref": "#/definitions/DataShareError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "DataAssetReferences_Get": {
+            "$ref": "./examples/DataAssetReferences_Get.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "DataAssetReference"
+        ],
+        "summary": "Adds a new DataAssetReference to a Proposal.",
+        "description": "Create a DataAssetReference in a Workspace ",
+        "operationId": "DataAssetReferences_Create",
```

Comment:
Why place dataAssetReferences underneath a proposal instead of in the body?  Are you expecting a lot of them or for it to mutate a lot from concurrent users?  Is there an rbac requirement?  They are very simple objects.

Reply:
over the lifetime of a proposal we expect it's child resources (DataAsset References, Scrip References, Invites, and Entitlements) to mutate quite a lot by multiple users - so we modeled all of them as such.